### PR TITLE
Add 3D stem plot

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This is a rebase and squash of #6271. Because the 3D projection changed since that was written, I had to throw away the original test images to prevent bloat, but I took that opportunity to drop some duplicates and use the mpl20 style. I then modernized docs to match current standards, and simplified some bits.

As this is completely new, I chose to avoid the optional things in the 2D `stem`, i.e., it does not attempt to guess coordinates for the baseline. It also always returns a `Line3DCollection` for `stemlines`.

I'm not entirely sure I like the way `zdir` works. It uses `juggle_axes`, so it does like other 2D-converted things and rotates what x/y/z mean. This makes sense for an artist that's a 2D plane, but this is 3D. On the other hand, `Line3D` works the same way. Perhaps the direction of stem lines/baseline should not be tied to `zdir`, but use `orientation` like in 2D?

I will write What's new when we decide what the right API is here.

Closes #6271.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [n/a] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
> I'm not entirely sure I like the way `zdir` works. It uses `juggle_axes`, so it does like other 2D-converted things and rotates what x/y/z mean. This makes sense for an artist that's a 2D plane, but this is 3D. On the other hand, `Line3D` works the same way. Perhaps the direction of stem lines/baseline should not be tied to `zdir`, but use `orientation` like in 2D?

On further thought, I think `zdir` may be the wrong approach, at least as used here. Though it was suggested in #6271, I don't think it does what we want. Namely, it decides, which way is 'up' for a 2D-to-3D conversion. But the parameters of `stem3d` are 3D already. Thus, using `zdir` rotates the result, and means you need to juggle around `x`/`y`/`z` if have an existing set and just want the stems in a different orientation starting from a different.

If we want to keep `zdir`, then I think it should apply only to the `bottom` baseline. Alternatively, we can have *orientation*={`x`, `y`, `z`}, which moves _only_ the baseline.

**Comment by QuLogic** [comment]
I changed this to use `orientation` instead of `zdir`, and tweaked the tests a bit.

**Comment by tacaswell** [comment]
I am :+1: in general, my primary concern is only being able to control the style via the plot-style format strings.  I think that this is OK to start with, but long term we may want to expose all of the knobs.

**Comment by tacaswell** [comment]
However, this matches the 2D stem (https://matplotlib.org/api/_as_gen/matplotlib.pyplot.stem.html) and you can always change the styling after the fact.

File: lib/mpl_toolkits/tests/test_mplot3d.py [hunk 1]
```diff
@@ -1203,6 +1203,41 @@ def test_errorbar3d():
...
+                            constrained_layout=True,
+                            subplot_kw={'projection': '3d'})
+    for ax, zdir in zip(axs, ['x', 'y', 'z']):
+        markerline, stemlines, baseline = ax.stem(
+            x, y, z,
+            linefmt='C4-.', markerfmt='C1D', basefmt='C2',
+            orientation=zdir)
+        ax.set_title(f'orientation={zdir}')
+        markerline.set(markerfacecolor='none', markeredgewidth=2)
+        baseline.set_linewidth(3)
```
Comment (anntzer) [inline lib/mpl_toolkits/tests/test_mplot3d.py:1238]: A single baseline image test should be enough?  e.g. using defaults on orientation="x" and customized formats on other orientations.

Comment (QuLogic) [reply lib/mpl_toolkits/tests/test_mplot3d.py:1232]: I squashed them into one image.

**Review by dopplershift — APPROVED** [review]

**Review by efiring — APPROVED** [review]
Looks ready to go.

**Pull request merged**